src/solvers/elasticity_solver.cpp

```diff
@@ -166,27 +158,20 @@ void ElasticitySolver::QuasiStaticSolve()
   // Apply the boundary conditions
   *bc_rhs_ = *rhs_;
   for (auto& bc : ess_bdr_) {
-    bc->vec_coef->SetTime(time_);
-    displacement_->gf->ProjectBdrCoefficient(*bc->vec_coef, bc->markers);
+    SLIC_ERROR_IF(!std::holds_alternative<std::shared_ptr<mfem::VectorCoefficient>>(bc.coef),
```

Comment:
I'll link this in the new PR.

The assertion itself doesn't need to be there, but if it ends up being `False`, the call to `std::get` on the next line will throw an exception.  I would argue that it's still preferable to the current approach with two variables, because if `vec_coef` remained uninitialized, you'd end up trying to dereference a nullptr.